app/src/main/java/hangbt/hust/androidbasic/MainActivity.java

```diff
@@ -3,12 +3,229 @@
 import androidx.appcompat.app.AppCompatActivity;
 
 import android.os.Bundle;
+import android.view.View;
+import android.widget.TextView;
+import android.widget.Toast;
 
 public class MainActivity extends AppCompatActivity {
 
+    TextView tv0, tv1, tv2, tv3, tv4, tv5, tv6, tv7, tv8, tv9, tv00, tv_cong, tv_tru, tv_nhan, tv_chia, tv_bang;
+    TextView tv_result;
+    int value1 = -1, value2 = -1;
+    boolean cong, tru, nhan ,chia;
+    String txtresult = "";
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        findId();
+
+        cong = false; tru = false; nhan = false; chia = false;
+
+        tv0.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                txtresult += "0";
+                tv_result.setText(txtresult);
+            }
+        });
+
+        tv1.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                txtresult += "1";
+                tv_result.setText(txtresult);
+            }
+        });
+
+        tv2.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                txtresult += "2";
+                tv_result.setText(txtresult);
+            }
+        });
+
+        tv3.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                txtresult += "3";
+                tv_result.setText(txtresult);
+            }
+        });
+
+        tv4.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                txtresult += "4";
+                tv_result.setText(txtresult);
+            }
+        });
+
+        tv5.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                txtresult += "5";
+                tv_result.setText(txtresult);
+            }
+        });
+
+        tv6.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                txtresult += "6";
+                tv_result.setText(txtresult);
+            }
+        });
+
+        tv7.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                txtresult += "7";
+                tv_result.setText(txtresult);
+            }
+        });
+
+        tv8.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                txtresult += "8";
```

Comment:
Không để hard code, khai báo trong hằng số 
```suggestion
txtresult += Numbers.EIGHT;
```